app/atomic_svc.py

```diff
@@ -322,3 +343,71 @@ async def _save_ability(self, entries, test):
             return True
 
         return False
+
+    async def _prereq_formater(self, prereq_test, prereq, prereq_type, exec_type, ability_command):
+        """
+        Format prereqs as a header test block for an ability
+        :param prereq_test: Test to see if the ability is required
+        :param prereq: Command to install prereq if required
+        :param prereq_type: Which executor this prereq should target (psh, sh, cmd)
+        :param exec_type: Which executor this ability should target (psh, sh, cmd)
+        :param ability_command: Existing commands for this ability
+        :return: Full formed, staged command
+        """
+        output = ""
+        prereq = prereq.rstrip()
+        if 'exit' not in prereq_test.lower() or prereq.startswith('echo "') or \
+                (prereq.startswith('echo ') and ('Run' in prereq or 'Sorry,' in prereq)):
+            if self.processing_debug:
+                self.log.debug(f'Action ({prereq}) cannot be automated automatically.')
+                if prereq.startswith('echo'):
+                    self.log.debug(f'Try to satisfy: {prereq.split("echo")[1].split("; exit")[0]}')
+                elif prereq.startswith('Write-Host'):
+                    self.log.debug(f'Try to satisfy: {prereq.split("Write-Host ")[1]}')
+            raise ExtractionError
+        if prereq_type == 'sh':
+            segments = prereq_test.split(';')
+            if 'exit 1' in segments[1]:
+                # check is "falsy"
+                output += f"{segments[0]}; then {prereq}; fi;"
+            else:
+                # check is "truthy"
+                output += f"{segments[0]}; then : ; else {prereq}; fi;"
+        elif prereq_type == 'psh':
+            if prereq_test.startswith('Try'):
+                temp = f"{prereq_test.replace('exit 1', prereq)}"
+                tstring = "Set-Variable -Name 'a' -Value 'pass'"
```

Comment:
is this just supposed to represent a no-op?

Reply:
Yep - it's easier to do no-ops in some languages that others, but this was the one that came up for powershell in google. If you know of a shorter one, I'm game for making it more straightforward.

Reply:
You should be able to have just blank space between the braces, or even just a semicolon. Something like:
```
if ($true) { } else {echo hi}
if ($true) { ; } else {echo hi}
```

Reply:
Nice, thanks for pointing that out.